### PR TITLE
OCPBUGS-22531: bump otel dependencies

### PR DESCRIPTION
**Pull request by simonpasquier**
Required to address CVE-2023-45142.

### CODE REVIEW

**Comment by simonpasquier** [comment]
/hold

Need to wait for the fix to land in the master branch.

File: go.mod [hunk 1]
```diff
@@ -55,12 +55,12 @@ require (
 	github.com/stretchr/testify v1.8.4
 	github.com/vultr/govultr/v2 v2.17.2
 	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.42.0
```
Comment (jan--f) [inline go.mod:57]:
```suggestion
	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.44.0
```
Should this dependency not also be updated?

Comment (simonpasquier) [reply go.mod:57]: Yes, good catch!

**Comment by simonpasquier** [comment]
Blocked by https://github.com/openshift/prometheus/pull/185

**Pull request merged**